pkg/oci/oci.go

```diff
@@ -106,7 +106,7 @@ func (s *RuleService) validateReference(ctx context.Context, ref name.Reference,
 	}
 
 	// verify image signature (optional)
-	if sv.SecretName != "" {
+	if sv.SecretName != "" || (sv.PublicKeys != nil && len(sv.PublicKeys) > 0) {
```

Comment:
nit: `len(sv.PublicKeys) > 0` would be sufficient here, even if `sv.PublicKeys` is `nil`